test/integration/test_elastic_torch.py

```diff
@@ -37,3 +39,18 @@ def test_all_hosts_blacklisted(self, mock_get_min_start_hosts):
     @mock.patch('horovod.runner.gloo_run._get_min_start_hosts', return_value=1)
     def test_min_hosts_timeout(self, mock_get_min_start_hosts):
         self.skipTest('This test fails due to https://github.com/horovod/horovod/issues/2030')
+
+    @mock.patch('horovod.runner.elastic.driver.DISCOVER_HOSTS_FREQUENCY_SECS', 0.01)
+    @mock.patch('horovod.runner.gloo_run._get_min_start_hosts', return_value=1)
+    def test_fault_tolerance_without_scaling(self, mock_get_min_start_hosts):
```

Comment:
we could further restrict this to torch head and see if it is exclusively an issue for >= `1.9.0`:

    if LooseVersion(torch.__version__) >= LooseVersion('1.9.0'):

Reply:
Sounds good!